spec/models/student_spec.rb

```diff
@@ -0,0 +1,57 @@
+require "rails_helper"
+
+RSpec.describe Student, type: :model do
+  context "validations" do
+    context "with all attributes" do
+      let(:student_with_all_attributes) { FactoryBot.build(:student) }
+
+      it "is valid" do
+        expect(student_with_all_attributes).to be_valid
+      end
+    end
+
+    context "with missing first_name" do
+      let(:student_without_first_name) { FactoryBot.build(:student, first_name: nil) }
+
+      it "is invalid" do
+        expect(student_without_first_name).not_to be_valid
+      end
+    end
+
+    # Add similar contexts for other attributes (last_name, email, birth_date, gender)
+
+    context "with invalid email format" do
+      let(:student_with_invalid_email) { FactoryBot.build(:student, email: "xyzemail") }
+
+      it "is invalid" do
+        expect(student_with_invalid_email).not_to be_valid
+        expect(student_with_invalid_email.errors[:email]).to include("is invalid")
+      end
+    end
+
+    context "with valid email format" do
+      let(:student_with_valid_email) { FactoryBot.build(:student, email: "xyzemail@gmail.com") }
+
+      it "is valid" do
+        expect(student_with_valid_email).to be_valid
+      end
+    end
+
+    context "with duplicate email" do
+      let!(:existing_student) { FactoryBot.create(:student) }
+      let(:student_with_duplicate_email) { FactoryBot.build(:student, email: existing_student.email) }
+
+      it "is invalid" do
+        expect(student_with_duplicate_email).not_to be_valid
+        expect(student_with_duplicate_email.errors[:email]).to include("has already been taken")
+      end
+    end
+  end
```

Comment:
This is missing specs for quite a few validations, such as last_name, birth_date, etc.

There's a common gem to simplify writing validation specs, we use it in Plus https://github.com/thoughtbot/shoulda-matchers

E.g.
```
describe 'validations' do
  it { should validate_presence_of(:first_name) }
  it { should_validate_uniqueness_of(:email) }
end
```

Reply:
@josh-cooper this is to demonstrate working of factory bot gem. Should I update it with shoulda-matchers? or just add it for missing validations